tiny_dnn/nodes.h

```diff
@@ -43,24 +43,32 @@ namespace cereal {
 
 template <typename Archive>
 void save(Archive & ar, const std::vector<tiny_dnn::layerptr_t>& v) {
+#ifndef CNN_NO_SERIALIZATION
     ar(cereal::make_size_tag((cereal::size_type)v.size()));
     for (auto n : v) {
         tiny_dnn::layer::save_layer(ar, *n);
     }
+#else
+    throw tiny_dnn::nn_error("TinyDNN was not build with Serialization support");
```

Comment:
Anyways, serialization functions can also be externally implemented with `cereal` library.
So, after moving serialization code outside the library, we don't need to use these preprocessor directives.